vendor/afragen/git-updater-lite/Lite.php

```diff
@@ -0,0 +1,415 @@
+				return;
+			}
+
+			if ( empty( $this->update_server ) || is_wp_error( $this->update_server ) ) {
+				return new \WP_Error( 'invalid_domain', 'Invalid update server domain', $this->update_server );
+			}
+			$url      = "$this->update_server/wp-json/git-updater/v1/update-api/?slug=$this->slug";
+			$response = get_site_transient( "git-updater-lite_{$this->file}" );
+			if ( ! $response ) {
+				$response = wp_remote_post( $url );
+				if ( is_wp_error( $response ) ) {
+					return $response;
+				}
+
+				$this->api_data = (object) json_decode( wp_remote_retrieve_body( $response ), true );
+				if ( null === $this->api_data || empty( (array) $this->api_data ) || property_exists( $this->api_data, 'error' ) ) {
+					return new \WP_Error( 'non_json_api_response', 'Poorly formed JSON', $response );
+				}
+				$this->api_data->file = $this->file;
+
+				/*
+				* Set transient for 5 minutes as AWS sets 5 minute timeout
+				* for release asset redirect.
+				*
+				* Set limited timeout so wp_remote_post() not hit as frequently.
+				* wp_remote_post() for plugin/theme check can run on every pageload
+				* for certain pages.
+				*/
+				set_site_transient( "git-updater-lite_{$this->file}", $this->api_data, 5 * \MINUTE_IN_SECONDS );
+			} else {
+				if ( property_exists( $response, 'error' ) ) {
+					return new \WP_Error( 'repo-no-exist', 'Specified repo does not exist' );
+				}
+				$this->api_data = $response;
+			}
+
+			$this->load_hooks();
+		}
+
+		/**
+		 * Load hooks.
+		 *
+		 * @return void
+		 */
+		public function load_hooks() {
+			$type = $this->api_data->type;
+			add_filter( 'upgrader_source_selection', array( $this, 'upgrader_source_selection' ), 10, 4 );
+			add_filter( "{$type}s_api", array( $this, 'repo_api_details' ), 99, 3 );
+			add_filter( "site_transient_update_{$type}s", array( $this, 'update_site_transient' ), 15, 1 );
+			if ( ! is_multisite() ) {
+				add_filter( 'wp_prepare_themes_for_js', array( $this, 'customize_theme_update_html' ) );
+			}
+
+			// Load hook for adding authentication headers for download packages.
+			add_filter(
+				'upgrader_pre_download',
+				function () {
+					add_filter( 'http_request_args', array( $this, 'add_auth_header' ), 15, 2 );
+					return false; // upgrader_pre_download filter default return value.
+				}
+			);
+		}
+
+		/**
+		 * Correctly rename dependency for activation.
+		 *
+		 * @param string                           $source        Path fo $source.
+		 * @param string                           $remote_source Path of $remote_source.
+		 * @param \Plugin_Upgrader|\Theme_Upgrader $upgrader      An Upgrader object.
+		 * @param array                            $hook_extra    Array of hook data.
+		 *
+		 * @return string|\WP_Error
+		 */
+		public function upgrader_source_selection( string $source, string $remote_source, \Plugin_Upgrader|\Theme_Upgrader $upgrader, $hook_extra = null ) {
```

Comment:
It would be great to remove the union typed typehint - I can't see any at the moment - so AspireUpdate can continue to support PHP 7.4 as that accounts for roughly 30% of WP sites for the moment. It's a minimal adjustment, not unlike some of the adjustments we made to dependencies when merging their updates to WP Core, and will be needed less and less in the future as sites move to PHP 8+.

Reply:
Made PHP 7.4 compatible in v2.2.0